mattermost-plugin/webapp/src/components/FocalboardUnfurl.tsx

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useState, useEffect} from 'react'
+import ReactMarkdown from 'react-markdown'
+
+const Avatar = (window as any).Components.Avatar
+const Timestamp = (window as any).Components.Timestamp
+const imageURLForUser = (window as any).Components.imageURLForUser
+
+import './FocalboardUnfurl.scss'
+
+type Props = {
+    embed: {
+        type: string,
+        data: string,
+    }
+}
+
+export const FocalboardUnfurl = (props: Props): JSX.Element => {
+    const focalboardInformation = JSON.parse(props.embed.data)
+    const workspaceID = focalboardInformation.workspaceID
+    const blockID = focalboardInformation.blockID
+    const baseURL = focalboardInformation.baseURL
+    const boardID = focalboardInformation.boardID
+    const viewID = focalboardInformation.viewID
+    const [card, setCard] = useState<{title?: string, type?: string, updateAt?: number, createdBy?: string, fields?: { icon: string, contentOrder: Array<string | string[]>, properties: {key?: string}}}>({})
+    const [content, setContent] = useState<string>('')
+    const [board, setBoard] = useState<{title?: string, fields?: { cardProperties: Array<unknown> }}>({})
+
+    if (!workspaceID || !blockID || !baseURL || !boardID || !viewID) {
+        return <></>
+    }
+
+    useEffect(() => {
+        const fetchData = async () => {
+            const response = await fetch(`${baseURL}/plugins/focalboard/api/v1/workspaces/${workspaceID}/blocks?block_id=${blockID}`, {
```

Comment:
Probably should be using a selector here, rather than calling the API endpoint directly.

Reply:
Spoke with Hossein, not available here, in plugin.